### PR TITLE
All CHAR_DATA names are now strings

### PR DESCRIPTION
**Pull request by mattgodbolt**
[not expecting a review, this is me staging this change for my own diffing/testing]

### CODE REVIEW

File: src/act_obj.cpp [hunk 1]
```diff
@@ -2313,7 +2308,7 @@ void do_hailcorpse(CHAR_DATA *ch, const char *argument) {
             continue;
 
         for (current_obj = current_place->contents; current_obj != nullptr; current_obj = current_obj->next_content) {
-            if ((current_obj->item_type == ITEM_CORPSE_PC) && strstr(current_obj->short_descr, ch->name) != nullptr) {
+            if ((current_obj->item_type == ITEM_CORPSE_PC) && matches_inside(ch->name, current_obj->short_descr)) {
```
Comment (snellers) [inline src/act_obj.cpp:2311]: This is actually interesting, it's a very similar problem to what I faced when working on the summoner - PC corpses don't include the player name in the 'name' field. I made the summoning routine match on short_descr just as this done.

File: src/act_wiz.cpp [hunk 2]
```diff
@@ -1278,23 +1272,14 @@ void do_mstat(CHAR_DATA *ch, const char *argument) {
...
-                     race_table[victim->race].name, victim->sex == 1 ? "male" : victim->sex == 2 ? "female" : "neutral",
-                     victim->in_room == nullptr ? 0 : victim->in_room->vnum)
-            < 0)
-            bug("Buffer too small at "
-                "_file_name_"
-                ":"
-                "1296"
-                " - message was truncated");
-    } while (0);
+    bug_snprintf(buf, sizeof(buf), "Vnum: %d  Format: %s  Race: %s  Sex: %s  Room: %d\n\r",
```
Comment (snellers) [inline src/act_wiz.cpp:1279]: eventually bug_snprintf() itself can be ditched I suspect

File: src/challeng.cpp [hunk 3]
```diff
@@ -426,9 +411,24 @@ void do_duel(CHAR_DATA *ch, const char *argument) {
     send_to_char("|cYou did not challenge that person.|w\n\r", ch);
 }
 
+namespace {
+void tell_results_to(int who) {
+    if (who == 0) {
```
Comment (snellers) [inline src/challeng.cpp:416]: I know you only moved it, but the who 0/1 thing is non-obvious.  We can redo it separately.

File: src/challeng.cpp [hunk 4]
```diff
@@ -565,34 +540,32 @@ int fighting_duel(CHAR_DATA *ch, CHAR_DATA *victim) {
...
 /* Checks to see that various things are correct.  Only input is
    toggle to say what to do on error.  0=nothing, 1=abort duel. */
 static int check_duel_status(int f) {
-    int flag = 0;
+    unsigned int flag = 0;
 
     if (challenger != nullptr) {
-        if (strncmp(challenger->name, challenger_name, NAME_SIZE))
-            flag = 1;
+        if (challenger->name == challenger_name)
```
Comment (snellers) [inline src/challeng.cpp:557]: is the logic on these == right?  `if(strncmp` will only return 0 if it's equal?

File: src/mob_prog.cpp [hunk 5]
```diff
@@ -1023,14 +1023,14 @@ void mprog_translate(char ch, char *t, CHAR_DATA *mob, const CHAR_DATA *actor, c
 
     *t = '\0';
     switch (ch) {
-    case 'i': one_argument(mob->name, t); break;
+    case 'i': one_argument(mob->name.c_str(), t); break;
```
Comment (snellers) [inline src/mob_prog.cpp:1026]: haha, minimal work on mob_prog.cpp I see, fair doos, that is another rat's nest and I'm pretty skeptical about it right now.

**Review by snellers — APPROVED** [review]
Looks  like a pretty mega step forward. One or two minor comments.

File: src/act_wiz.cpp (hunk 2 above)
Comment (mattgodbolt) [reply src/act_wiz.cpp:1279]: Absolutely!

File: src/challeng.cpp (hunk 3 above)
Comment (mattgodbolt) [reply src/challeng.cpp:416]: Totally: an `enum class` or similar...or anything...

File: src/mob_prog.cpp (hunk 5 above)
Comment (mattgodbolt) [reply src/mob_prog.cpp:1026]: 100% I suspect it's not long for this world so I didn't spend too much time making it better!

File: src/challeng.cpp (hunk 4 above)
Comment (mattgodbolt) [reply src/challeng.cpp:557]: Wow! Great spot: I was 100% wrong here!!!

File: src/mob_prog.cpp (hunk 5 above)
Comment (snellers) [reply src/mob_prog.cpp:1026]: Do I get a free spell or a customized item? :)

File: src/challeng.cpp (hunk 4 above)
Comment (mattgodbolt) [reply src/challeng.cpp]: Fixed! Am going to go over everything one more time then I'll merge!

**Pull request merged**